openpgp/write.go

```diff
@@ -435,27 +443,30 @@ type signatureWriter struct {
 	signer        *packet.PrivateKey
 	sigType       packet.SignatureType
 	config        *packet.Config
+	metadata      *packet.LiteralData // Protected by v5 keys only
 }
 
 func (s signatureWriter) Write(data []byte) (int, error) {
 	s.wrappedHash.Write(data)
-	flag := 0
 	switch s.sigType {
 	case packet.SigTypeBinary:
 		return s.literalData.Write(data)
 	case packet.SigTypeText:
+		flag := 0
 		return writeCanonical(s.literalData, data, &flag)
 	}
 	return 0, errors.UnsupportedError("unsupported signature type: " + strconv.Itoa(int(s.sigType)))
 }
 
 func (s signatureWriter) Close() error {
 	sig := &packet.Signature{
+		Version:      s.signer.Version,
 		SigType:      s.sigType,
 		PubKeyAlgo:   s.signer.PubKeyAlgo,
 		Hash:         s.hashType,
 		CreationTime: s.config.Now(),
 		IssuerKeyId:  &s.signer.KeyId,
+		Metadata:      s.metadata,
```

Comment:
The final nitpick, not even @twiss can beat me: there is a space too much :joy:

Reply:
Here, and indulgence to grant you pardon:
https://upload.wikimedia.org/wikipedia/commons/thumb/4/4e/Indulgence_San_Giovanni_in_Laterano_2006-09-07.jpg/1920px-Indulgence_San_Giovanni_in_Laterano_2006-09-07.jpg